ironfish-cli/src/commands/start.ts

```diff
@@ -223,7 +230,28 @@ export default class Start extends IronfishCommand {
     }
     this.log(` `)
 
-    await NodeUtils.waitForOpen(node, () => this.closing)
+    let walletPassphrase: string | undefined
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      try {
+        await NodeUtils.waitForOpen(node, () => this.closing, walletPassphrase)
+        break
+      } catch (e) {
+        if (e instanceof EncryptedWalletMigrationError) {
+          this.logger.info(e.message)
+          walletPassphrase = await inputPrompt(
+            'Enter your passphrase to unlock the wallet',
+            true,
+            {
+              password: true,
+            },
+          )
+        } else {
+          throw e
+        }
+      }
+    }
```

Comment:
Not thrilled about doing this in the `start` command or about passing the passphrase down through `NodeUtils`. Alternatively we could catch the error here and tell the user to run `ironfish migrations:start` and enter their passphrase there

Reply:
I would much prefer to do that than add the password flow here to start.

Mainly because if someone has their wallet encrypted unfortunately they just need to run migrations manually. I don't like the idea of adding interactive elements to the start command.